src/OSPSuite.Presentation/Presenters/Importer/ColumnMappingPresenter.cs

```diff
@@ -732,6 +733,22 @@ private void invalidateErrorUnit()
          if (measurementColumn == null) return;
 
          //either both measurement and error units should be coming from excel columns, or they should have the same dimension
+         checkErrorUnitAgainstMeasurementUnit(errorColumn, measurementColumn);
+
+         //if the unit is already automatically set but the Standard Deviation not,
+         //Then set the STD to Arithmetic
+         setDeviationWhenUnitNotNull(errorColumn);
+      }
+
+      private static void setDeviationWhenUnitNotNull(Column errorColumn)
+      {
+         if (errorColumn.Unit.SelectedUnit != null && errorColumn.Unit.SelectedUnit != UnitDescription.InvalidUnit
+                                                   && errorColumn.ErrorStdDev == null)
+            errorColumn.ErrorStdDev = Constants.STD_DEV_ARITHMETIC;
+      }
+
+      private static void checkErrorUnitAgainstMeasurementUnit(Column errorColumn, Column measurementColumn)
+      {
          if (errorColumn.ErrorStdDev != Constants.STD_DEV_GEOMETRIC &&
              ((errorColumn.Unit?.ColumnName.IsNullOrEmpty() != measurementColumn.Unit?.ColumnName.IsNullOrEmpty()) ||
```

Comment:
I feel that we are missing a few helper methods here that would make those lines more readable. It feels like we are never sure if the unit is going to bet set, of it's going to be equal to something.

I am glad we have tests to that refactoring should not ber hard